infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -709,18 +733,19 @@ public Map<LogReplicationSession, ReplicationStatus> queryReplicationStatus() {
             return replicationStatusMap;
         }
 
-        if (topologyDescriptor.getLocalClusterDescriptor().getRole() != ClusterRole.SOURCE &&
-                topologyDescriptor.getLocalClusterDescriptor().getRole() != ClusterRole.SINK) {
-            log.error("Replication Status Query in incorrect role {}", topologyDescriptor.getLocalClusterDescriptor().getRole());
+        if (!isSource() && !isSink()) {
+            log.error("Received Replication Status Query in Incorrect Role, cluster is neither SOURCE/SINK");
             return replicationStatusMap;
         }
 
         return sessionManager.getReplicationStatus();
     }
 
+
     public UUID forceSnapshotSync(LogReplicationSession session) throws LogReplicationDiscoveryServiceException {
-        if(topologyDescriptor.getLocalClusterDescriptor().getRole() == ClusterRole.SINK) {
-            String errorStr = "Force snapshot sync not supported on sink cluster.";
+        if(!topologyDescriptor.getRemoteSinkClusters().containsKey(session.getSinkClusterId())) {
```

Comment:
This is not enough, right? for all you know the passes session in forceSnapshotSync API could to have local cluster as source, that should be verified first.

Reply:
Sorry, I did not get this..

Reply:
hmm okay, I see your point. Will push a fix soon